adminapi/dataset/__init__.py

```diff
@@ -437,7 +420,32 @@ def create(
     }
 
     response = send_request(CREATE_ENDPOINT, request)
-    qs = Query({'hostname': attributes['hostname']})
-    qs._handle_response(response)
+    if response['status'] == 'error':
+        _handle_exception(response)
+
+    return _format_server(response['result'][0])
+
+
+def _format_server(result):
+    object_id = result['object_id']
+    server_obj = ServerObject([], object_id)
+
+    for attribute_id, value in list(result.items()):
+        if isinstance(value, list):
+            casted_value = MultiAttr(
+                (_format_attribute_value(v) for v in value),
+                server_obj,
+                attribute_id,
+            )
+        else:
+            casted_value = _format_attribute_value(value)
+
+        dict.__setitem__(server_obj, attribute_id, casted_value)
+
+    return server_obj
+
 
-    return qs.get()
+def _format_attribute_value(value):
+    if isinstance(value, dict):
+        return _format_server(value)
```

Comment:
Possible endless recursion (not really common in that case, but theoretically possible.